src/qililab/digital/circuit_transpiler.py

```diff
@@ -37,11 +39,17 @@ class CircuitTranspiler:
     - ``transpile_circuit``: runs both of the methods above sequentially
 
     Args:
-        platform (Platform): platform object containing the runcard and the chip's physical qubits.
+        settings (DigitalCompilationSettings | Platform): Object containing the digital compilations settings and the info on chip's physical qubits.
     """
 
-    def __init__(self, digital_compilation_settings: DigitalCompilationSettings):  # type: ignore # ignore typing to avoid importing platform and causing circular imports
-        self.digital_compilation_settings = digital_compilation_settings
+    def __init__(self, settings: DigitalCompilationSettings | Any):  # type: ignore # ignore typing to avoid importing platform and causing circular imports
+        if isinstance(settings, DigitalCompilationSettings):
+            self.digital_compilation_settings = settings
+        else:
+            try:
+                self.digital_compilation_settings = settings.digital_compilation_settings
+            except AttributeError as e:
+                raise ValueError("`setting`s must be a `DigitalCompilationSettings` or `Platform`.") from e
```

Comment:
Why do you allow passing settings as `Platform`? `DigitalCompilationSettings` has everything you need, no?

Reply:
Yess!

But I think, for users that want to build the `Transpiler` to do routing/transpiling manually, its going to be much easier if they can just instantiate it with the platform they are used to.

Since they will never construct only a `DigitalCompilationSettings`, but always the `Platform`, and then they would need to pass the `platform.digital_compilation_settings`... So I accepted both, and select it for them :)

Reply:
**Two more things:**
1) At the start, I was just correcting, only having the `Platform` -> The `DigitalCompilationSettings`, and then I decided to keep both. But if we don't want that, at least, we need to change `Platform` -> `DigitalCompilationSettings` here.
2) And the `Any`, comes from `Platform` causing circular import error :)

Reply:
The general principle is that classes should only accept parameters in their `__init__` method that are essential for their operation. For instance, if someone wants to use the `Transpiler` independently—such as to compile a circuit and inspect the result—they shouldn’t be forced to build the entire `Platform` object. Instead, they should only need the relevant settings. If a `Platform` object is already available, its `platform.digital_compilation_settings` can simply be passed as the parameter, as you correctly mentioned.

The circular import issue you encountered is also a sign that the class hierarchy needs improvement. When `Platform` imports `Transpiler` and `Transpiler` imports `Platform`, it creates a problematic dependency cycle. To work around this, you're resorting to using `Any`, which is also discouraged and should only be used in exceptional cases (e.g., when dealing with truly dynamic types that we don't know in compile time).

While renaming the parameter to `settings` is perfectly fine, please revert the rest of the changes to the previous state. 🙏

Reply:
You could also rename the class attribute to just `settings`.

Reply:
> they shouldn’t be forced to build the entire Platform object.
I mean, but they aren't, because I was accepting both `Platform` and the `DigitalCompilationSettings`, so they are not forced :)

> you're resorting to using `Any`, which is also discouraged and should only be used in exceptional cases 
That I agree with, it was an ugly quick solution, which I shouldn't have done

Reply:
> You could also rename the class attribute to just settings.

Like the idea, doing it right now 👌

Reply:
@fedonman Comming back to the discussion about accepting only `DigitalCompilationSettings` or both that and `Platfom`, the difference for the user in the instantiation of a `Transpiler` will be this:

With both (right now):
```python
transpiler = Transpiler(platform)
routed_c, layout = transpiler.route_circuit(c, ...)
```
vs, what you are asking:
```python
transpiler = Transpiler(platform.digital_compilation_settings)
routed_c, layout = transpiler.route_circuit(c, ...)
```
if you think it's okey anyway, I'll change it tomorrow 👍

Reply:
Yes, please change back to `DigitalCompilationSettings` only.

Reply:
Done! 🔥